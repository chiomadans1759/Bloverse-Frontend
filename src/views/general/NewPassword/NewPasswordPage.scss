@import 'sass-rem';

.new-password-container {
  padding-top: rem(15px);
  padding-bottom: rem(20px);
  background-color: #333333;
  min-height: 100vh;
  color: white;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  background-attachment: fixed;
  background-image:
        linear-gradient(rgba(37, 42, 48, 0.2), rgba(37, 42, 48, 0.2)),
        url('../../../assets/signin-image.jpg');
}

.new-password-content {
  min-height: calc(100vh - 170px);
  position: relative;
}

.new-password-intro-one {
  font-size: rem(24px);
  font-weight: 900;
  margin-bottom: rem(8px);
  width: rem(281px);
}

.new-password-intro-two {
  font-size: rem(16px);
  margin-bottom: rem(24px);
  width: rem(281px);
}

.new-password-form-section {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}
