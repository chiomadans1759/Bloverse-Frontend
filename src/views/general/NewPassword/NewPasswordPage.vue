<template>
  <div class="new-password-container">
    <div class="top">
      <span class="top-left-container">
        <a href="/">
          <i class="fal fa-long-arrow-left fa-2x"></i>
          <span class="back-text">Back</span>
        </a>
        <router-link to="/" class="router-link">
          <img class="logo" src="@/assets/Bloverse-white.svg">
          <img class="blue-logo" src="@/assets/Bloverse.svg">
        </router-link>
      </span>
    </div>
    <div class="new-password-content d-flex flex-column justify-content-center align-items-center">
      <div>
        <h4 class="new-password-intro-one">New Password</h4>
        <h5 class="new-password-intro-two">Enter your new password below</h5>

        <div class="new-password-form-section">
          <text-input
            id="password"
            name="password"
            type="password"
            placeholder="Password"
            :value="password"
            :onChange="onChange"
            custom-class="text-input"
          />

          <text-input
            id="confirmPassword"
            name="confirmPassword"
            type="password"
            placeholder="Confirm Password"
            :value="password"
            :onChange="onChange"
            custom-class="text-input"
          />
          <Button customClass="btn btn-primary password-primary-button">Save</Button>
        </div>
      </div>
    </div>
    <div class="password-bottom">
      <p>
        <span class="bloverse">Bloverse</span>
        {{year}}. All Right reserved
      </p>
    </div>
  </div>
</template>

<script>
import Button from "@/components/Button/Button.vue";
import TextInput from "@/components/TextInput/TextInput.vue";

export default {
  data() {
    return {
      email: "",
      year: new Date().getFullYear()
    };
  },
  components: {
    Button,
    TextInput
  },
  methods: {
    onChange: function(event) {
      this[event.target.name] = event.target.value;
    }
  }
};
</script>

<style lang="scss">
@import "./NewPasswordPage.scss";
</style>
