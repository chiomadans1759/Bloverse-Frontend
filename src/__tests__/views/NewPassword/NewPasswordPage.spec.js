import { mount, RouterLinkStub } from '@vue/test-utils'
import NewPassword from '@/views/general/NewPassword/NewPasswordPage.vue';

describe('NewPassword Page', () => {
  global.hideTawkBot = jest.fn();
  
  it('should render correctly', () => {

    const wrapper = mount(NewPassword, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.find(RouterLinkStub).props().to).toBe('/')
    expect(wrapper.element).toMatchSnapshot();
  });


  it('should update the component state when input is changed', () => {
    const wrapper = mount(NewPassword, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    const passwordInput = wrapper.find('#password');
    passwordInput.element.value = '1234567890';
    passwordInput.trigger('change');
    expect(wrapper.vm.password).toBe('1234567890');

    const confirmPasswordInput = wrapper.find('#confirmPassword');
    confirmPasswordInput.element.value = '1234567890';
    confirmPasswordInput.trigger('change');
    expect(wrapper.vm.confirmPassword).toBe('1234567890');
  });
});
