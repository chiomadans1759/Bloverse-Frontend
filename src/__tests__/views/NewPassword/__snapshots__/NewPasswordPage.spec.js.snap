// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPassword Page should render correctly 1`] = `
<div
  class="new-password-container"
>
  <div
    class="top"
  >
    <span
      class="top-left-container"
    >
      <a
        href="/"
      >
        <i
          class="fal fa-long-arrow-left fa-2x"
        />
         
        <span
          class="back-text"
        >
          Back
        </span>
      </a>
       
      <a
        class="router-link"
      >
        <img
          class="logo"
          src="@/assets/Bloverse-white.svg"
        />
         
        <img
          class="blue-logo"
          src="@/assets/Bloverse.svg"
        />
      </a>
    </span>
  </div>
   
  <div
    class="new-password-content d-flex flex-column justify-content-center align-items-center"
  >
    <div>
      <h4
        class="new-password-intro-one"
      >
        New Password
      </h4>
       
      <h5
        class="new-password-intro-two"
      >
        Enter your new password below
      </h5>
       
      <div
        class="new-password-form-section"
      >
        <form>
          <div
            class="form-group"
          >
            <input
              class="form-control defaultClass text-input"
              id="password"
              name="password"
              placeholder="Password"
            />
          </div>
        </form>
         
        <form>
          <div
            class="form-group"
          >
            <input
              class="form-control defaultClass text-input"
              id="confirmPassword"
              name="confirmPassword"
              placeholder="Confirm Password"
            />
          </div>
        </form>
         
        <div>
          <button
            class="btn btn-primary password-primary-button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="password-bottom"
  >
    <p>
      <span
        class="bloverse"
      >
        Bloverse
      </span>
      
      2019. All Right reserved
    
    </p>
  </div>
</div>
`;
